File: codingame/community/sumOfDivisors/sumOfDivisors.scala
import math._
import scala.util._
import BigInt._

object Solution extends App {
    val n = readInt
    var x:BigInt = n;
    var y:BigInt = x/2;
    var sum:BigInt = y;
    
    for ( i <- 1 until y) {
        sum+= y/i*(1+x/i);
        if (x/i%2==1 )
            sum+=1+y/i;    
    }
    
    sum+=3;
    println(sum)
}

File: codingame/community/sumOfDivisors/sumOfDivisors.py
import sys
import math
x = int(raw_input())
y = int(x/2)
z = 0
i = 1
while i <= x:
    z += y//i * (1+ x//i)
    if (x//i%2 == 1 ):
        z += 1+y//i
    i += 1
print(z)

